FWCore/Framework/test/stubs/ToyModules.cc

```diff
@@ -361,11 +362,47 @@ namespace edmtest {
     e.put(std::make_unique<Prodigal>(parent->value));
   }
 
+  class IntProductFilter : public edm::global::EDFilter<> {
+  public:
+    explicit IntProductFilter(edm::ParameterSet const& p)
+        : token_(consumes(p.getParameter<edm::InputTag>("label"))),
+          threshold_(p.getParameter<int>("threshold")),
+          shouldProduce_(p.getParameter<bool>("shouldProduce")) {
+      if (shouldProduce_) {
+        putToken_ = produces<edmtest::IntProduct>();
+      }
+    }
+
+    bool filter(edm::StreamID, edm::Event& iEvent, edm::EventSetup const&) const {
+      auto const& product = iEvent.get(token_);
+      if (product.value < threshold_) {
+        return false;
+      }
+      iEvent.emplace(putToken_, product);
+      return true;
+    }
+
+    static void fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+      edm::ParameterSetDescription desc;
+      desc.add<edm::InputTag>("label");
+      desc.add<int>("threshold", 0);
+      desc.add<bool>("shouldProduce", false);
+      descriptions.addDefault(desc);
+    }
+
+  private:
+    const edm::EDGetTokenT<edmtest::IntProduct> token_;
+    edm::EDPutTokenT<edmtest::IntProduct> putToken_;
+    int threshold_;
```

Comment:
This could be `const` (since other members, that can be, are)?
```suggestion
    const int threshold_;
```

Reply:
done